peer.py

```diff
@@ -44,18 +50,29 @@ def make_handshake(self):
 
     async def send_handshake(self):
         try:
+            self.connection_try += 1 # increment the number of times we tried to connect to this peer
             self.reader, self.writer = await asyncio.open_connection(
                 self.peer_ip, self.peer_port
             )
             self.writer.write(self.make_handshake())
             print(f"[{self.peer_ip}]: SENT HANDSHAKE")
             await self.writer.drain()
             await self.validate_handshake()
-        except:
+        except ConnectionRefusedError:
```

Comment:
how this works is we first initialize the `PeerConnection` instance with a `self.connection_try` being 0 to indicate we haven't connected to it. On lines 62-70 then we try to connect to it five times, and if it fails we remove it on the peer list